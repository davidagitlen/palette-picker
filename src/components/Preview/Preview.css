.Preview {
  box-shadow: 3px 3px 7px #A9A9A9;
  font-family: 'Roboto', sans-serif;
  width: 23%;
}

.page {
  border: solid #808080 1px;
  height: 375px;
}

.page-header {
  height: 50px;
  font-size: 28px;
  line-height: 50px;
  margin: 0 auto;
}

.page-profile {
  float: left;
  height: 60px;
  margin: 0px 10px 0px 10px;
  width: 75px;
}

.page-title {
  font-size: 21px;
  line-height: 30px;
  margin: 5px 2px;
}

.page-body {
  font-size: 12px;
  margin: 2px 2px 10px 10px;
  text-align: left;
}

p:last-child {
  padding: 0 0 15px 0;
}

.page-footer {
  height: 24px;
  margin: 0 auto;
}