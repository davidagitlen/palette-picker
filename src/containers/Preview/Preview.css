.Preview {
  font-family: 'Roboto', sans-serif;
  width: 23%;
}

.page {
  border: solid #E8E8E8 2px;
  height: 375px;
}

.page-header {
  height: 50px;
  font-size: 36px;
  line-height: 36px;
  margin: 0 auto;
}

.page-png {
  float: left;
  height: 60px;
  margin: 0px 10px 0px 10px;
  width: 75px;
}

.page-title {
  font-size: 28px;
  margin: 5px 2px;
}

.page-body {
  font-size: 12px;
  margin: 2px 2px 10px 10px;
  text-align: left;
}

.page-footer {
  height: 52px;
  margin: 0 auto;
}