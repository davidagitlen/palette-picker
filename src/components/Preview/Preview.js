import React from 'react';
import '../Preview/Preview.css';
import profile from '../../images/profile.png';

const Preview = (props) => {
  const { projectName, paletteName, colors } = props;
  const header = colors.colors[0].hex_1;
  const title = colors.colors[1].hex_2;
  const border = `3px solid ${colors.colors[2].hex_3}`;
  const body = colors.colors[3].hex_4;
  const footer = colors.colors[4].hex_5;
  return (
    <div className="Preview">
      <div className="page">
        <header className="page-header" style={{ backgroundColor: header }}>
          {projectName || "Project Name"}
        </header>
        <div>
          <h4 className="page-title" style={{ color: title }}>
            {paletteName || "Palette Name"}
          </h4>
          <img
            className="page-profile"
            src={profile}
            alt="stock png"
            style={{ border: border }}
          />
          <p className="page-body" style={{ color: body }}>
            Lorem lean startup ipsum product market fit customer development
            acquihire technical cofounder. User engagement A/B testing shrink a
            market venture capital pitch deck.
          </p>
          <p className="page-body" style={{ color: body }}>
            Social bookmarking group buying crowded market pivot onboarding
            freemium prototype ping pong. Supply chain bandwidth holy grail
            disruptive series. Analytics burn rate entrepreneur beta hackathon
            agile development.
          </p>
          <p className="page-body" style={{ color: body }}>
            Tech virtual drone online browser platform through in a system. But
            stream software offline. Professor install angel sector anywhere
            create at components smart. Document fab developers encryption
            smartphone powered, bespoke blockstack edit atoms. Privacy news data
            policies analytics documents.
          </p>
        </div>
        <footer
          className="page-footer"
          style={{ backgroundColor: footer }}
        ></footer>
      </div>
    </div>
  );
}

export default Preview;