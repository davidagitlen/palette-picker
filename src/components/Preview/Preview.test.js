import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Preview', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
    <Preview 
      paletteName='TestPalette'
      projectName='TestProject'
      colors={{ colors: [
        { hex_1: '#ABABAB' },
        { hex_2: '#ABABAB' },
        { hex_3: '#ABABAB' },
        { hex_4: '#ABABAB' },
        { hex_5: '#ABABAB' }
      ]
      }}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});