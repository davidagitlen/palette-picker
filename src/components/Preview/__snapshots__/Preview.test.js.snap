// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview should match the snapshot with all data passed in correctly 1`] = `
<div
  className="Preview"
>
  <div
    className="page"
  >
    <header
      className="page-header"
      style={
        Object {
          "backgroundColor": "#ABABAB",
        }
      }
    >
      TestProject
    </header>
    <div>
      <h4
        className="page-title"
        style={
          Object {
            "color": "#ABABAB",
          }
        }
      >
        TestPalette
      </h4>
      <img
        alt="stock png"
        className="page-profile"
        src="profile.png"
        style={
          Object {
            "border": "3px solid #ABABAB",
          }
        }
      />
      <p
        className="page-body"
        style={
          Object {
            "color": "#ABABAB",
          }
        }
      >
        Lorem lean startup ipsum product market fit customer development acquihire technical cofounder. User engagement A/B testing shrink a market venture capital pitch deck.
      </p>
      <p
        className="page-body"
        style={
          Object {
            "color": "#ABABAB",
          }
        }
      >
        Social bookmarking group buying crowded market pivot onboarding freemium prototype ping pong. Supply chain bandwidth holy grail disruptive series. Analytics burn rate entrepreneur beta hackathon agile development.
      </p>
      <p
        className="page-body"
        style={
          Object {
            "color": "#ABABAB",
          }
        }
      >
        Tech virtual drone online browser platform through in a system. But stream software offline. Professor install angel sector anywhere create at components smart. Document fab developers encryption smartphone powered, bespoke blockstack edit atoms. Privacy news data policies analytics documents.
      </p>
    </div>
    <footer
      className="page-footer"
      style={
        Object {
          "backgroundColor": "#ABABAB",
        }
      }
    />
  </div>
</div>
`;
